src/app/api/staff/mark-story/[staff_pick_id]/route.ts

```diff
@@ -0,0 +1,99 @@
+import prisma from "@/lib/prisma";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { patchSchema } from "../schema";
+
+interface Params {
+  staff_pick_id: string;
+}
+export async function DELETE(req: NextRequest, { params }: { params: Params }) {
+  try {
+    const { staff_pick_id } = params;
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+
+    if (!user || !user.roles.includes("EDITOR")) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+    const staffPickOld = await prisma.staffPick.findUnique({
+      where: { id: staff_pick_id },
+    });
+    if (!staffPickOld) {
+      return NextResponse.json({ error: "Not found" }, { status: 404 });
+    }
+
+    const deleteStaffPickPromise = prisma.staffPick.delete({
+      where: { id: staff_pick_id },
+    });
+    //create record
+    const createRecordPromise = prisma.staffPickRecord.create({
+      data: {
+        updateType: "DELETE",
+        storyId: staffPickOld.storyId,
+        staffId: user.id,
+      },
+    });
+    await Promise.all([deleteStaffPickPromise, createRecordPromise]);
+    return NextResponse.json({}, { status: 200 });
+  } catch (err) {
+    console.error(err);
+    return NextResponse.json(
+      { error: "Internal Server Error" },
+      { status: 500 },
+    );
+  }
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: Params },
+) {
+  try {
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+    const { staff_pick_id } = params;
```

Comment:
same question as the previous file, is it possible to return status 400 if staff_pick_id is null/undefined/an unexpected type?

Reply:
I will use the schema to check if id is the valid type. If not a valid type, it will return 400